### PR TITLE
feat: add starknetkit hook and return connector object instead of wallet object when connecting with starknetkit

### PR DESCRIPTION
**Pull request by bluecco**
Make `starknetkit` modal to be compatible with `starknet-react`

- `connect` method now returns a `Connector` instead of a `StarknetwWindowObject`. this means dapps will have to call `.connect()` on their side if they're using `starknetkit` without `starknet-react`

example snippet for plain `starknetkit` without `starknet-react`:
```typescript
const connector = await connect({
    modalMode: "neverAsk",
    webWalletUrl,
    argentMobileOptions: {
      dappName: "Example dapp",
      chainId: argentMobileChainId,
    },
  })

  await connector?.connect()

  windowStarknet = connector?.wallet as StarknetWindowObject | null
```

- add `useStarknetkitConnectors` to expose: `connectors` and `connect` (starknetkit modal)  

example snippet:
```typescript
import { goerli } from "@starknet-react/chains"
import {
  useConnect,,
  publicProvider,
} from "@starknet-react/core"

export const availableConnectors = [
  new InjectedConnector({ options: { id: "argentX" } }),
  new InjectedConnector({ options: { id: "braavos" } }),
  new ArgentMobileConnector({
    dappName: "Example dapp",
    chainId: argentMobileChainId,
  }),
  new WebWalletConnector({ url: webWalletUrl }),
]

const App = () => {
 const chains = [goerli]
  const providers = publicProvider()
  return  (
    <StarknetConfig
      chains={chains}
      provider={providers}
      connectors={availableConnectors as Connector[]}
    >
      <Content />
    </StarknetConfig>
  )
}

const Content = () => {
  const { connect } = useConnect()
  const { starknetkitConnect } = useStarknetkitConnectors({
    argentMobileOptions: {
      dappName: "Example dapp",
      chainId: argentMobileChainId,
    },
    webWalletUrl,
  })

  return (
    <button 
      onClick={async () => {
         const connector = await starknetkitConnect()
         if (connector) {
           connect({ connector })
         }
      }}>Connect</button>
  )
}
```

### CODE REVIEW

**Comment by gergold** [comment]
2 general comments:

(1) I think we shouldn't break the current flow for existing non-starknet-react developers

(2) I was expecting an higher abstraction level where we could have a `StarknetkitConnector` conforming to the `Connector` interface. This is probably more work to achieve that. I'ml saying that because here in your snippet, you have to define twice the StarknetKit options... Saying that I've another idea, as a compromise, would that work in the `Content()` function?
```
const { connectors, connect } = useConnect();
const { starknetkitConnect } = useStarknetkitConnectors({ connectors });
...
```
So basically you define the connectors only once, and then you inject them into `useStarknetkitConnectors`

**Comment by bluecco** [comment]
1) 
it would be a minor (breaking) change, having the modal to return `Connector` or `StarknetWindowObject` would still require checks on dapps side (type checking the result with some guards)

we could:
- conditionally call the `.connect()` method in the callback ih `main.ts`, depending on how the connect method is invoked (maybe adding a property, like `autoConnect`) https://github.com/argentlabs/starknetkit/blob/799743bdd24b5f378fd6dd1c28f4a07f2d07517a/src/main.ts#L94
- return an object like { connector, wallet } (wallet is defined with the `autoConnect`)



2) the snippet you wrote works perfectly with the current solution
```typescript
const { connectors, connect } = useConnect();
const { starknetkitConnect } = useStarknetkitConnectors({ connectors });
```


`useStarknetkitConnects` has the same options as `connect` defined in main (since `starknetkitConnect` is a wrapper to that function) since we have to show the `Modal.svelte` modal

```typescript

**Comment by vladutjs** [comment]
@gergold @bluecco I think it's fine if we have a minor breaking change, if in the future we have a better and more consistent API, and aligned with what the community expects. It's a very young library, should we care about fully backwards compatibility so early? They can also use a fix version if they want to stay at the old API.

**Comment by bluecco** [comment]
@vladutjs @gergold I think too it's a minor breaking change but I also get @gergold point

that's why I was proposing something like a `autoConnect` property (maybe a different name like `type` with types `Connector | StarknetWindowObject`, since `autoConnect` could cause confusion) and having the `connect` method returning 2 types


```typescript
export interface ConnectOptions extends GetWalletOptions {
  argentMobileOptions?: ArgentMobileConnectorOptions
  dappName?: string
  connectors?: Connector[]
  modalMode?: "alwaysAsk" | "canAsk" | "neverAsk"
  modalTheme?: "light" | "dark" | "system"
  storeVersion?: StoreVersion | null
  webWalletUrl?: string
  type: Connector | StarknetWindowObject
}
```

and the callback would be something like (mostly pseudo code for `if` condition)

```typescript
callback: async (connector: Connector | StarknetWindowObject | null) => {
          try {
            selectedConnector = connector

            if (connector !== null && connector.id !== "argentWebWallet") {
              setStarknetLastConnectedWallet(connector.id)
            }

           // check type of connector
           if (type === Connector) {
             resolve(connector)
           } else {
              await connector?.connect()
              resolve(connector?.wallet ?? null)
            }
          } finally {
            setTimeout(() => modal.$destroy())
          }
        },

**Review by vladutjs — APPROVED** [review]
